ansible-wazuh-agent/tasks/Linux.yml

```diff
@@ -10,60 +10,138 @@
+        {% if wazuh_agent_authd.ssl_auto_negotiate == 'yes' %}-a{% endif %}
+      register: agent_auth_output
+      when:
+        - check_keys.stat.size == 0
+        - wazuh_managers.0.address is not none
+      tags:
+        - config
+        - authd
+
+    - name: Linux | Verify agent registration
+      shell: echo {{ agent_auth_output }} | grep "Valid key created"
+      when:
+        - check_keys.stat.size == 0
+        - wazuh_managers.0.address is not none
+      tags:
+        - config
+        - authd
+
+  when: wazuh_agent_authd.enable == true
+
+- name: Linux | Agent registration via rest-API
+  block:
+
+    - name: Retrieving rest-API Credentials
+      include_vars: api_pass.yml
+      tags:
+        - config
+        - api
+
+    - name: Linux | Create the agent key via rest-API
+      uri:
+        url: "{{ wazuh_managers.0.api_proto }}://{{ wazuh_managers.0.address }}:{{ wazuh_managers.0.api_port }}/agents/"
+        validate_certs: no
+        method: POST
+        body: {"name":"{{ inventory_hostname }}","ip":"{{ ansible_default_ipv4.address }}"}
+        body_format: json
+        status_code: 200
+        headers:
+          Content-Type: "application/json"
+        user: "{{ wazuh_managers.0.api_user }}"
+        password: "{{ api_pass }}"
+      register: newagent_api
+      changed_when: newagent_api.json.error == 0
+      when:
+        - check_keys.stat.size == 0
+        - wazuh_managers.0.address is not none
+      delegate_to: localhost
+      become: no
+      tags:
+        - config
+        - api
+
+    - name: Linux | Retieve the client key via rest-API
+      uri:
+        url: "{{ wazuh_managers.0.api_proto }}://{{ wazuh_managers.0.address }}:{{ wazuh_managers.0.api_port }}/agents/{{ newagent_api.json.data.id }}/key"
+        validate_certs: no
+        method: GET
+        return_content: yes
+        user: "{{ wazuh_managers.0.api_user }}"
+        password: "{{ api_pass }}"
+      when:
+        - check_keys.stat.size == 0
+        - wazuh_managers.0.address is not none
+        - newagent_api.json.error == 0
+      register: newagentkey_api
+      delegate_to: localhost
+      become: no
+      tags:
+        - config
+        - api
+
+    - name: Linux | Register agent (via rest-API)
+      command: /var/ossec/bin/manage_agents
+      environment:
+        OSSEC_ACTION: i
+        OSSEC_AGENT_NAME: '{{ inventory_hostname }}'
+        OSSEC_AGENT_IP: '{{ ansible_default_ipv4.address }}'
+        OSSEC_AGENT_ID: '{{ newagent_api.json.data.id }}'
+        OSSEC_AGENT_KEY: '{{ newagentkey_api.json.data }}'
+        OSSEC_ACTION_CONFIRMED: y
+      register: manage_agents_output
+      when:
+        - check_keys.stat.size == 0
+        - wazuh_managers.0.address is not none
+        - newagent_api.changed
+      tags:
+        - config
+        - api
+      notify: restart wazuh-agent
+
+  when: wazuh_agent_authd.enable == false and ( wazuh_managers.0.api_user is defined and api_pass is defined )
```

Comment:
Hi @angystardust,

It looks like this condition will never be true meanwhile the `api_pass` var remains inside the same block, just be checking the value of `wazuh_agent_authd.enable` should be good, for example:

>   when: wazuh_agent_authd.enable == false